### PR TITLE
Add name attribute to application scope

### LINKED ISSUE #269: Using image labels in aqua_application_scopes does not work due to missing name attribute
Hi there,

Using image labels in `aqua_application_scopes` does not work due to missing `name` attribute in `Variables`.

### Terraform Version
The version should not matter as it is an issue with the provider.

### Affected Resource(s)
Please list the resources as a list, for example:
- aquasec_application_scope

### Terraform Configuration Files
```hcl
resource "aquasec_application_scope" "my_scope" {
  name        = "MyScope"
  description = "Resources in MyScope"
  categories {
    artifacts {
      image {
        expression = "v1"
        variables {
          attribute = "image.label"
          value = "my.aqua.scope.MyImageLabel"
        }
      }
    }
  }
}
```

### Expected Behavior
When creating a application scope using Docker Labels via the console, querying the `api/v2/access_management/scopes/MyScope` will return the following result.

```json
{
    "name": "MyScope",
    "categories": {
        "artifacts": {
            "image": {
                "expression": "v1",
                "variables": [
                    {
                        "attribute": "image.label",
                        "value": "MyImageLabelValue",
                        "name": "my.image.label"
                    }
                ]
            },
            "function": {
                "expression": "",
                "variables": []
            },
            "cf": {
                "expression": "",
                "variables": []
            }
        },
        "workloads": {
            "os": {
                "expression": "",
                "variables": []
            },
            "kubernetes": {
                "expression": "",
                "variables": []
            },
            "cf": {
                "expression": "",
                "variables": []
            }
        },
        "infrastructure": {
            "kubernetes": {
                "expression": "",
                "variables": []
            },
            "os": {
                "expression": "",
                "variables": []
            }
        },
        "entity_scope": {
            "expression": "v1",
            "variables": [
                {
                    "attribute": "application_scope.name",
                    "value": "MyScope"
                }
            ]
        }
    },
    "description": "",
    "author": "me@home.com",
    "updated_at": "2024-07-18T14:49:18.498184Z",
    "owner_email": ""
}
```

### Actual Behavior
When creating a application scope using Docker Labels via Terraform, querying the `api/v2/access_management/scopes/MyScope` will return the following result.

This leads to filtering not being applied correctly, meaning containers with the specified label will not be included in the application scope.

```json
{
    "name": "MyScope",
    "categories": {
        "artifacts": {
            "image": {
                "expression": "v1",
                "variables": [
                    {
                        "attribute": "image.label",
                        "value": "my.image.label.MyImageLabelValue",
                    }
                ]
            },
            "function": {
                "expression": "",
                "variables": []
            },
            "cf": {
                "expression": "",
                "variables": []
            }
        },
        "workloads": {
            "os": {
                "expression": "",
                "variables": []
            },
            "kubernetes": {
                "expression": "",
                "variables": []
            },
            "cf": {
                "expression": "",
                "variables": []
            }
        },
        "infrastructure": {
            "kubernetes": {
                "expression": "",
                "variables": []
            },
            "os": {
                "expression": "",
                "variables": []
            }
        },
        "entity_scope": {
            "expression": "v1",
            "variables": [
                {
                    "attribute": "application_scope.name",
                    "value": "MyScope"
                }
            ]
        }
    },
    "description": "",
    "author": "me@home.com",
    "updated_at": "2024-07-18T14:49:18.498184Z",
    "owner_email": ""
}
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

1. Create an application scope using Docker labels via Terraform 
2. Go to Aqua console and try filtering in any option using the newly created application scope

### PR DESCRIPTION
**Pull request by Toqn**
fixes #269

### CODE REVIEW

**Comment by ehardisty** [comment]
Hi @Toqn, 

I've run the tests against your branch and there are some failures. The ones related to the Saas user resource are unrelated and will be fixed in #274 so don't worry about those. But it does appear your changes have failures for the application scope resource.

I'm aware this PR has been open for a long time now and just letting you know, we haven't forgotten about it. I have recently taken on a maintainer role for the terraform provider here at Aqua so will be working on implementing these changes and will keep you updated. I don't expect you to do anything to fix these issues.

Cheers
Emerson

**Comment by ehardisty** [comment]
OK sorry for the noise. This should be good to go as soon as #274 gets merged, we can then merge those changes into your branch to get a fully passing test suite. Will have these additions in the next release of the aquasec provider.

**Review by semyonmor — APPROVED** [review]

**Pull request merged**
